### PR TITLE
Set Router Language

### LINKED ISSUE #186: Set default routing language based on device locale
Investigate KM vs MI as well.

### PR DESCRIPTION
**Pull request by sarahlensing**
- Updates `OnTheRoad` dependency
- Adds method to set the `MapzenRoutingController`'s locale
- Uses default locale if locale not explicitly set
- Updates routing example to allow changing the router's language
- Adds tests to `MapzenRoutingController`

Closes #186

### CODE REVIEW

File: SampleApp/RoutingViewController.swift [hunk 1]
```diff
@@ -48,41 +68,76 @@ class RoutingViewController: UIViewController, RoutingSearchDelegate {
...
+    self.navigationItem.rightBarButtonItem = btn
+  }
 
-    var startingPoint = OTRRoutingPoint(coordinate: OTRGeoPointMake(40.7444892, -73.9900082), type: .break)
-    if let location = LocationManager.sharedManager.currentLocation {
-      startingPoint = OTRRoutingPoint(coordinate: OTRGeoPointMake(location.coordinate.latitude, location.coordinate.longitude), type: .break)
-    }
+  @objc private func changeRouterLanguage() {
+    let actionSheet = UIAlertController.init(title: "Router Language", message: "Choose a language", preferredStyle: .actionSheet)
+    actionSheet.addAction(UIAlertAction.init(title: "English", style: .default, handler: { [unowned self] (action) in
```
Comment (karimnaaji) [inline SampleApp/RoutingViewController.swift]: Wouldn't something like that work?
```swift
let languageIdByActionSheetTitle = [
    "English": "en_US",
    "French": "fr_FR",
    ...
]
for (actionTitle, languageIdentifier) in languageIdByActionSheetTitle {
    actionSheet.addAction(UIAlertAction.init(title: actionTitle, style: .default, handler: { [unowned self] (action) in
        self.routingLocale = Locale.init(identifier: languageIdentifier)
    }))
}
```

File: SampleApp/RoutingViewController.swift [hunk 2]
```diff
@@ -48,41 +68,76 @@ class RoutingViewController: UIViewController, RoutingSearchDelegate {
...
-                                                  self.currentRouteResult = routingResult
-                                                  self.routeResultTable?.show(routingResult!)
+    guard let currentLocation = LocationManager.sharedManager.currentLocation, let destination = destination else { return }
+    let startingPoint = OTRRoutingPoint(coordinate: OTRGeoPointMake(currentLocation.coordinate.latitude, currentLocation.coordinate.longitude), type: .break)
+    let endingPoint = OTRRoutingPoint(coordinate: destination, type: .break)
+
+    _ = routingController.requestRoute(withLocations: [startingPoint, endingPoint],
+                                       costingModel: .auto,
+                                       costingOption: nil,
+                                       directionsOptions: ["units" : "miles" as NSObject]) { (routingResult, token, error) in
```
Comment (karimnaaji) [inline SampleApp/RoutingViewController.swift]: Any plan in supporting an alternative `km` unit?

File: SampleApp/RoutingViewController.swift (hunk 1 above)
Comment (sarahlensing) [inline SampleApp/RoutingViewController.swift]: Ah yes, thats more concise. Updated.

File: SampleApp/RoutingViewController.swift (hunk 2 above)
Comment (sarahlensing) [inline SampleApp/RoutingViewController.swift]: Yes, the next step will be adding support for a units enum: https://github.com/mapzen/on-the-road_ios/issues/27

**Pull request merged**